engine/physics/src/physics/physics_2d.cpp

```diff
@@ -654,39 +654,42 @@ namespace dmPhysics
         {
             fixture = fixture->GetNext();
         }
-        assert(fixture != 0x0);
         return fixture;
     }
 
-    static inline b2GridShape* GetGridShape(b2Body* body, uint32_t index)
-    {
-        b2Fixture* fixture = GetFixture(body, index);
-        assert(fixture->GetShape()->GetType() == b2Shape::e_grid);
-        return (b2GridShape*) fixture->GetShape();
-    }
-
-    void SetGridShapeHull(HCollisionObject2D collision_object, uint32_t shape_index, uint32_t row, uint32_t column, uint32_t hull, HullFlags flags)
+    bool SetGridShapeHull(HCollisionObject2D collision_object, uint32_t shape_index, uint32_t row, uint32_t column, uint32_t hull, HullFlags flags)
     {
         b2Body* body = (b2Body*) collision_object;
-        b2GridShape* grid_shape = GetGridShape(body, shape_index);
+        b2Fixture* fixture = GetFixture(body, shape_index);
+        if (fixture == 0)
+        {
+            return false;
+        }
+        b2GridShape* grid_shape = (b2GridShape*) fixture->GetShape();
```

Comment:
You removed the "GetGridShape()" which actually checks the type.
How do you know this is a grid shape?

Reply:
Oh crap, didn't notice this. Will update!